sortedm2m_tests/test_field.py

```diff
@@ -183,6 +183,17 @@ def get_ids(queryset):
             return [obj.id for obj in queryset]
         self.assertEqual(get_ids(shelf.books.all()), get_ids(books))
 
+    def test_base_class_m2mprint(self):
+        shelf = self.model.objects.create()
+        shelf.books.add(self.books[0])
+
+        shelf_book = shelf.books.through
+
+        if hasattr(shelf_book, '__unicode__'):
+            self.assertEqual(shelf_book.__unicode__, m2mprint.__unicode__)
+        else:
+            self.assertTrue(True)
```

Comment:
I have not understood your last comment very well.
You have to write a test which checks the output of the `__str__` method is what you expect. No need to add if/else blocks.